From mtosatti@redhat.com  Thu Sep  3 14:33:18 2009
From: Avi Kivity <avi@qumranet.com>
Date: Thu,  6 Aug 2009 14:39:44 -0300
Subject: KVM: Allocate guest memory as MAP_PRIVATE, not MAP_SHARED
To: stable@kernel.org
Cc: avi@redhat.com
Message-ID: <1249580407-21883-5-git-send-email-mtosatti@redhat.com>


From: Avi Kivity <avi@qumranet.com>

(cherry picked from commit acee3c04e8208c17aad1baff99baa68d71640a19)

There is no reason to share internal memory slots with fork()ed instances.

Signed-off-by: Avi Kivity <avi@qumranet.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 arch/x86/kvm/x86.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -4003,7 +4003,7 @@ int kvm_arch_set_memory_region(struct kv
 			userspace_addr = do_mmap(NULL, 0,
 						 npages * PAGE_SIZE,
 						 PROT_READ | PROT_WRITE,
-						 MAP_SHARED | MAP_ANONYMOUS,
+						 MAP_PRIVATE | MAP_ANONYMOUS,
 						 0);
 			up_write(&current->mm->mmap_sem);
 
